### PR TITLE
Adding beta tag

### PR DESCRIPTION
**Pull request by AlvinChacko**
#11252

### CODE REVIEW

**Comment by AlvinChacko** [comment]
#build

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java [hunk 1]
```diff
@@ -225,6 +227,46 @@ public String getProperty(String key) {
         return value;
     }
 
+    /**
```
Comment (FlynnIBM) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: Fix this comment's grammar/spacing

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java [hunk 2]
```diff
@@ -225,6 +227,46 @@ public String getProperty(String key) {
...
+    }
+
+    /**
+     * Checks if the current version is in early access and then sets the BETA_EDITION_JVM_PROPERTY. It will
+     * be set to true if it found the early access tag else false.
+     *
+     */
+    public static void isAnyProductBeta() {
+        try {
+            System.out.println("Inside isAnyProductBeta " + System.getProperty(BETA_EDITION_JVM_PROPERTY));
```
Comment (FlynnIBM) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: Remove system.outs from function

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java [hunk 3]
```diff
@@ -225,6 +227,41 @@ public String getProperty(String key) {
...
+            isBeta = productEdition.equals(EARLY_ACCESS);
+        }
+        return isBeta;
+    }
+
+    /**
+     * Sets the com.ibm.ws.beta.edition JVM property to true if the Liberty version is EARLY_ACCESS,
+     * otherwise sets it to false.
+     */
+    public static void isAnyProductBeta() {
```
Comment (tspewak) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: The method name isAnyProductBeta seems odd.  I was expecting it to return something. Its purpose seems to be to ensure the java property is set.

File: dev/com.ibm.ws.kernel.boot_fat/fat/src/com/ibm/ws/kernel/boot/ServerStartJVMOptionsTest.java [hunk 4]
```diff
@@ -564,6 +564,79 @@ public void testStartServerCmdlineJVMOptionsAndSystemProps() throws Exception {
...
+                }
+
+                reader.close();
+                inputstream.close();
+            }
+        }
+
+        zipFile.close();
+        dumpFile.delete();
+        assertTrue("The jvm option was not found", foundTest1);
```
Comment (tspewak) [inline dev/com.ibm.ws.kernel.boot_fat/fat/src/com/ibm/ws/kernel/boot/ServerStartJVMOptionsTest.java]: would be better if it was   "JVM option com.ibm.ws.beta.edition was not found"

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java [hunk 5]
```diff
@@ -225,6 +227,41 @@ public String getProperty(String key) {
...
+        return isBeta;
+    }
+
+    /**
+     * Sets the com.ibm.ws.beta.edition JVM property to true if the Liberty version is EARLY_ACCESS,
+     * otherwise sets it to false.
+     */
+    public static void setBetaEditionJVMProperty() {
+        try {
+            if (System.getProperty(BETA_EDITION_JVM_PROPERTY) == null) {
```
Comment (tevans78) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: So what happens if the user has set this system property to "YES"? Please can you add a `public static boolean getBetaEdition()` method for internal components to use to get this value really easily? They shouldn't need to worry about what the exact value of the system property is or wether they need to call the method inside a doPriv or even use `ThreadIdentityManager.runAsServer()`.

Comment (tevans78) [reply dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: We should also socialize this at the next EOI, partly to make sure everyone knows that it exists but also to check that it forefills their requirements and that they don't end up writing their own code again.

Comment (mmerali96) [reply dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: Added the static method getBetaEdition() to ProductInfo in latest commit.

File: dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2012 IBM Corporation and others.
+d * Copyright (c) 2012, 2020 IBM Corporation and others.
```
Comment (FlynnIBM) [inline dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/productinfo/ProductInfo.java]: Ditch the d

**Comment by AlvinChacko** [comment]
#run-libby-bot

**Comment by mmerali96** [comment]
#build

**Comment by mmerali96** [comment]
#build

**Comment by mmerali96** [comment]
#build

**Comment by mmerali96** [comment]
#run-libby-bot

**Comment by mmerali96** [comment]
Tested these changes by adding a test to com.ibm.ws.kernel.boot_fat StartAndStopServerJVMOptions to make sure that beta edition property was set correctly on server startup.

**Pull request merged**